File: sdkwork-backend-sdk-csharp/Api/ChatMessageApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class ChatMessageApi
    {
        private readonly HttpClient _client;

        public ChatMessageApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update an existing chat message
        /// </summary>
        public async Task<PlusApiResultPlusChatMessageContentVO?> UpdateAsync(PlusChatMessageContentForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusChatMessageContentVO>(ApiPaths.BackendPath("/message"), body);
        }

        /// <summary>
        /// Create a new chat message
        /// </summary>
        public async Task<PlusApiResultPlusChatMessageContentVO?> CreateAsync(PlusChatMessageContentForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusChatMessageContentVO>(ApiPaths.BackendPath("/message"), body);
        }

        /// <summary>
        /// Update an existing chat message
        /// </summary>
        public async Task<PlusApiResultPlusChatMessageVO?> UpdateMessageAsync(PlusChatMessageForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusChatMessageVO>(ApiPaths.BackendPath("/chat/message"), body);
        }

        /// <summary>
        /// Create a new chat message
        /// </summary>
        public async Task<PlusApiResultPlusChatMessageVO?> CreateMessageAsync(PlusChatMessageForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusChatMessageVO>(ApiPaths.BackendPath("/chat/message"), body);
        }

        /// <summary>
        /// Get chat messages by page
        /// </summary>
        public async Task<PlusApiResultPagePlusChatMessageContentVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusChatMessageContentVO>(ApiPaths.BackendPath("/message/list"), body, query);
        }

        /// <summary>
        /// Get all chat messages
        /// </summary>
        public async Task<PlusApiResultListPlusChatMessageContentVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusChatMessageContentVO>(ApiPaths.BackendPath("/message/list/all"), body);
        }

        /// <summary>
        /// Get chat messages by page
        /// </summary>
        public async Task<PlusApiResultPagePlusChatMessageVO?> LoadMoreAsync(ChatMessageQueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusChatMessageVO>(ApiPaths.BackendPath("/chat/message/load_more"), body, query);
        }

        /// <summary>
        /// Get chat messages by page
        /// </summary>
        public async Task<PlusApiResultPagePlusChatMessageVO?> CreateListByPageAsync(ChatMessageQueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusChatMessageVO>(ApiPaths.BackendPath("/chat/message/list"), body, query);
        }

        /// <summary>
        /// Get all chat messages
        /// </summary>
        public async Task<PlusApiResultListPlusChatMessageVO?> CreateListAllEntitiesAsync(ChatMessageQueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusChatMessageVO>(ApiPaths.BackendPath("/chat/message/list/all"), body);
        }

        /// <summary>
        /// Get chat message by ID
        /// </summary>
        public async Task<PlusApiResultPlusChatMessageContentVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusChatMessageContentVO>(ApiPaths.BackendPath($"/message/{id}"));
        }

        /// <summary>
        /// Delete a chat message
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/message/{id}"));
        }

        /// <summary>
        /// Get a chat message by ID
        /// </summary>
        public async Task<PlusApiResultPlusChatMessageVO?> GetByIdMessageAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusChatMessageVO>(ApiPaths.BackendPath($"/chat/message/{id}"));
        }

        /// <summary>
        /// Delete a chat message
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteMessageAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/chat/message/{id}"));
        }
    }
}

File: sdkwork-backend-sdk-swift/Sources/API/ChatMessageApi.swift
import Foundation

public class ChatMessageApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update an existing chat message
    public func update(body: PlusChatMessageContentForm) async throws -> PlusApiResultPlusChatMessageContentVO? {
        let response = try await client.put(ApiPaths.backendPath("/message"), body: body)
        return response as? PlusApiResultPlusChatMessageContentVO
    }

    /// Create a new chat message
    public func create(body: PlusChatMessageContentForm) async throws -> PlusApiResultPlusChatMessageContentVO? {
        let response = try await client.post(ApiPaths.backendPath("/message"), body: body)
        return response as? PlusApiResultPlusChatMessageContentVO
    }

    /// Update an existing chat message
    public func updateMessage(body: PlusChatMessageForm) async throws -> PlusApiResultPlusChatMessageVO? {
        let response = try await client.put(ApiPaths.backendPath("/chat/message"), body: body)
        return response as? PlusApiResultPlusChatMessageVO
    }

    /// Create a new chat message
    public func createMessage(body: PlusChatMessageForm) async throws -> PlusApiResultPlusChatMessageVO? {
        let response = try await client.post(ApiPaths.backendPath("/chat/message"), body: body)
        return response as? PlusApiResultPlusChatMessageVO
    }

    /// Get chat messages by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusChatMessageContentVO? {
        let response = try await client.post(ApiPaths.backendPath("/message/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusChatMessageContentVO
    }

    /// Get all chat messages
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusChatMessageContentVO? {
        let response = try await client.post(ApiPaths.backendPath("/message/list/all"), body: body)
        return response as? PlusApiResultListPlusChatMessageContentVO
    }

    /// Get chat messages by page
    public func loadMore(body: ChatMessageQueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusChatMessageVO? {
        let response = try await client.post(ApiPaths.backendPath("/chat/message/load_more"), body: body, params: params)
        return response as? PlusApiResultPagePlusChatMessageVO
    }

    /// Get chat messages by page
    public func createListByPage(body: ChatMessageQueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusChatMessageVO? {
        let response = try await client.post(ApiPaths.backendPath("/chat/message/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusChatMessageVO
    }

    /// Get all chat messages
    public func createListAllEntities(body: ChatMessageQueryListForm? = nil) async throws -> PlusApiResultListPlusChatMessageVO? {
        let response = try await client.post(ApiPaths.backendPath("/chat/message/list/all"), body: body)
        return response as? PlusApiResultListPlusChatMessageVO
    }

    /// Get chat message by ID
    public func getById(id: String) async throws -> PlusApiResultPlusChatMessageContentVO? {
        let response = try await client.get(ApiPaths.backendPath("/message/\(id)"))
        return response as? PlusApiResultPlusChatMessageContentVO
    }

    /// Delete a chat message
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/message/\(id)"))
        return response as? PlusApiResultBoolean
    }

    /// Get a chat message by ID
    public func getByIdMessage(id: String) async throws -> PlusApiResultPlusChatMessageVO? {
        let response = try await client.get(ApiPaths.backendPath("/chat/message/\(id)"))
        return response as? PlusApiResultPlusChatMessageVO
    }

    /// Delete a chat message
    public func deleteMessage(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/chat/message/\(id)"))
        return response as? PlusApiResultBoolean
    }
}
